Source/WebKit/Platform/IPC/Connection.h

```diff
@@ -805,7 +807,7 @@ Connection::AsyncReplyHandler Connection::makeAsyncReplyHandler(typename T::Prom
                 producer.reject(Error::FailedToDecodeReplyArguments);
```

Comment:
I still think this work should be done in the promise queue.
So if only possible, it'd be great to chain the promise so that the decoder would run on the promise queue.
E.g. the type of `DirectDispatchAsyncReplyHandler` would be `CompletionHandler<UniqueRef<Decoder>>`

Reply:
I agree, doing any work outside the IPC queue would be preferable.
Issue is at the time the promise is resolved/created ; it may not have a listener yet, and so there would be no promise queue usable yet. The queue/handler is attached when someone calls `whenSettled`/`then` on the promise.

I've redone all this and instead defined a `sendWithPromisedReplyOnDispatcher` which takes a dispatcher as argument

But you give me a great idea, we could just resume the promise with the decoder, which would not decode yet, but once whenSettled *is* called , then we would run the decoder.

It does require additional backend to NativePromise though so I will take this as a future challenge :)